src/components/MeasureRoutes/MeasureRoutes.test.tsx

```diff
@@ -0,0 +1,153 @@
+const useMeasureServiceApiMock =
+  useMeasureServiceApi as jest.Mock<MeasureServiceApi>;
+
+const MEASURE_CREATEDBY = "testuser@example.com";
+const measure = {
+  id: "measure ID",
+  createdBy: MEASURE_CREATEDBY,
+} as Measure;
+
+const serviceApiMock = {
+  fetchMeasure: jest.fn().mockResolvedValue(measure),
+  fetchMeasures: jest.fn().mockResolvedValue({
+    content: [measure],
+    totalPages: 1,
+    totalElements: 1,
+    numberOfElements: 1,
+    pageable: {
+      sort: {
+        empty: false,
+        sorted: true,
+        unsorted: false,
+      },
+      offset: 0,
+      pageNumber: 0,
+      pageSize: 10,
+      paged: true,
+      unpaged: false,
+    },
+  }),
+} as unknown as MeasureServiceApi;
+
+useMeasureServiceApiMock.mockImplementation(() => {
+  return serviceApiMock;
+});
+
+jest.mock("@madie/madie-util", () => ({
+  useOktaTokens: () => ({
+    getAccessToken: () => "test.jwt",
+    getUserName: () => "TestUser@example.com", //#nosec
+  }),
+  measureStore: {
+    state: null,
+    initialState: null,
+    updateMeasure: (measure) => measure,
+  },
+}));
+
+const serviceConfig: ServiceConfig = {
+  measureService: {
+    baseUrl: "example-service-url",
+  },
+  elmTranslationService: {
+    baseUrl: "test-elm-service",
+  },
+};
+
+jest.mock("../notfound/NotFound", () => () => {
+  return (
+    <div data-testid="notfound-component-mocked">404 NotFound Component</div>
+  );
+});
+
+// react no op error is caused by two awaits in one it call. ignorable.
+describe("Measure Router", () => {
+  test("Render a default notFound page", async () => {
+    await act(async () => {
+      const { findByTestId } = await render(
+        <ApiContextProvider value={serviceConfig}>
+          <MemoryRouter
+            initialEntries={[
+              {
+                pathname: "",
+                search: "",
+                hash: "",
+                state: undefined,
+                key: "1fewtg",
+              },
+            ]}
+          >
+            <MeasureBrowserRoutes />
+          </MemoryRouter>
+        </ApiContextProvider>
+      );
+      const notFound = await findByTestId("notfound-component-mocked");
+      expect(notFound).toBeInTheDocument();
+    });
+  });
+
+  test("Router routes to measureLanding", async () => {
+    await act(async () => {
+      const { findByTestId } = await render(
+        <ApiContextProvider value={serviceConfig}>
+          <MemoryRouter
+            initialEntries={[
+              {
+                pathname: "/measures",
+                search: "",
+                hash: "",
+                state: undefined,
+                key: "1fewtg",
+              },
+            ]}
+          >
+            <MeasureRoutes />
+          </MemoryRouter>
+        </ApiContextProvider>
+      );
+      const measureLanding = await findByTestId("measure-landing");
+      expect(measureLanding).toBeInTheDocument();
+    });
+  });
+
+  test("Router routes to EditMeasure", async () => {
```

Comment:
This test thows the following warning:
>  Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.

Is this the "react no op error" referenced on line 80 or something unrelated?